mesonbuild/backend/ninjabackend.py

```diff
@@ -1033,9 +1033,16 @@ def generate_single_java_compile(self, src, target, compiler, outfile):
         args += self.build.get_project_args(compiler, target.subproject)
         args += target.get_java_args()
         args += compiler.get_output_args(self.get_target_private_dir(target))
+        if mesonlib.is_windows():
+            separator = ';'
+        else:
+            separator = ':'
+        sourcepath = ''
```

Comment:
This is os.pathsep

Reply:
ah, great, I guessed there must be something like that but didn't spend long enough looking. thanks.